### PR TITLE
freepascal: Add version 3.0.4

### LINKED ISSUE #844: [Request] Add FreePascal
`extras` bucket has `Lazarus` which is graphical IDE tool based on free pascal programming language. But often required to build a library without GUI interaction and this is impossible, since bare pascal is not available on scoop.

I believe [FreePascal](https://www.freepascal.org/) could make very nice addition to scoop's `main` bucket.

### PR DESCRIPTION
**Pull request by shytikov**
- Closes #844 

Providing definition file that will allow scoop install `freepascal` compiler (FPC)

### CODE REVIEW

File: bucket/freepascal.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{
+    "homepage": "https://www.freepascal.org",
+    "version": "3.0.4",
+    "description": "Professional Pascal compiler.",
+    "license": {
+        "identifier": "GPL-2.0-or-later | FPC-modified-LGPL-2.0-or-later",
+        "url": "http://wiki.lazarus.freepascal.org/licensing"
+    },
+    "url": "https://sourceforge.net/projects/freepascal/files/Win32/3.0.4/fpc-3.0.4.i386-win32.exe/download",
+    "hash": "1f6323d3362eca1f0b2f63160c2329205df0415f6aa9d8e42910629aaaa823bb",
```
Comment (Ash258) [inline bucket/freepascal.json]:
```suggestion
    "url": "https://downloads.sourceforge.net/project/freepascal/Win32/3.0.4/fpc-3.0.4.i386-win32.exe",
    "hash": "sha1:69b0d42739ccb22108a35d8f108741a7883ff3ff",
```

File: bucket/freepascal.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    ],
+    "shortcuts": [
+        [
+            "bin\\i386-win32\\fp.exe",
+            "Free Pascal IDE"
+        ]
+    ],
+    "checkver": "fpc-(?<fpc>[\\d.]+).i386-win32.exe",
+     "autoupdate": {
+        "url": "https://sourceforge.net/projects/freepascal/files/Win32/$version/fpc-$version.i386-win32.exe/download"
```
Comment (Ash258) [inline bucket/freepascal.json]:
```suggestion
    "autoupdate": {
        "url": "https://downloads.sourceforge.net/project/freepascal/Win32/$version/fpc-$version.i386-win32.exe"
```

File: bucket/freepascal.json [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    "hash": "1f6323d3362eca1f0b2f63160c2329205df0415f6aa9d8e42910629aaaa823bb",
+    "innosetup": true,
+    "bin": [
+        "bin\\i386-win32\\fp.exe",
+        "bin\\i386-win32\\fpc.exe"
+    ],
+    "shortcuts": [
+        [
+            "bin\\i386-win32\\fp.exe",
+            "Free Pascal IDE"
```
Comment (Ash258) [inline bucket/freepascal.json]:
```suggestion
            "Free Pascal IDE",
            "",
            "bin\\i386-win32\\fp32.ico"
```

**Comment by rashil2000** [comment]
@shytikov can you update this to the latest version?

**Comment by Valinor** [comment]
> @shytikov can you update this to the latest version?

shytikov delete his repository so I make a new PR, update and correct checkver and autoupdate. Seem working, see #2872

**Pull request closed without merging**